wagtail/search/checks.py

```diff
@@ -0,0 +1,25 @@
+from django.core.checks import Warning, register
+
+
+@register('search')
+def page_search_fields_check(app_configs, **kwargs):
+    """Checks each page model with search_fields to core fields are included"""
+    from wagtail.core.models import Page, get_page_models
+
+    page_models = get_page_models()
+    errors = []
+
+    for cls in page_models:
+        if not all(field in cls.search_fields for field in Page.search_fields[:10]):
```

Comment:
nitpick: would be great to have a note as to why this only looks at the first 10 search fields from base `Page`

Reply:
Good find, I put a note about it in the PR but have now added some clarity in a comment.